Source/BoundaryConditions/WarpXEvolvePML.cpp

```diff
@@ -56,25 +57,92 @@ WarpX::DampPML (int lev, PatchType patch_type)
             const Box& tby  = mfi.tilebox(By_nodal_flag);
             const Box& tbz  = mfi.tilebox(Bz_nodal_flag);
 
-            WRPX_DAMP_PML(tex.loVect(), tex.hiVect(),
-			    tey.loVect(), tey.hiVect(),
-			    tez.loVect(), tez.hiVect(),
-			    tbx.loVect(), tbx.hiVect(),
-    			tby.loVect(), tby.hiVect(),
-	    		tbz.loVect(), tbz.hiVect(),
-		    	BL_TO_FORTRAN_3D((*pml_E[0])[mfi]),
-			    BL_TO_FORTRAN_3D((*pml_E[1])[mfi]),
-			    BL_TO_FORTRAN_3D((*pml_E[2])[mfi]),
-			    BL_TO_FORTRAN_3D((*pml_B[0])[mfi]),
-			    BL_TO_FORTRAN_3D((*pml_B[1])[mfi]),
-			    BL_TO_FORTRAN_3D((*pml_B[2])[mfi]),
-			    WRPX_PML_TO_FORTRAN(sigba[mfi]));
+            auto const& pml_Exfab = pml_E[0]->array(mfi);
+            auto const& pml_Eyfab = pml_E[1]->array(mfi);
+            auto const& pml_Ezfab = pml_E[2]->array(mfi);
+            auto const& pml_Bxfab = pml_B[0]->array(mfi);
+            auto const& pml_Byfab = pml_B[1]->array(mfi);
+            auto const& pml_Bzfab = pml_B[2]->array(mfi);
+            auto const& AMREX_RESTRICT sigma_fac_x = sigba[mfi].sigma_fac[0].data();
+#if (AMREX_SPACEDIM == 3) 
+            auto const& AMREX_RESTRICT sigma_fac_y = sigba[mfi].sigma_fac[1].data();
+            auto const& AMREX_RESTRICT sigma_fac_z = sigba[mfi].sigma_fac[2].data();
+#else
+            Real* sigma_fac_y ;
```

Comment:
Would it make sense to explicitly have `Real* sigma_fac_y = nullptr;` here?

Reply:
probably safer, it's used way further down and passed directly in a call.
Also, maybe add `AMREX_RESTRICT`?

Is there actually a reason to make this a pointer and not a normal variable? Does this refer to a scalar value?

Reply:
They are not scalar.  I think `auto const& AMREX_RESTRICT sigma_fac_z =` should be `amrex::Real const * AMREX_RESTRICT sigma_fac_z = `.

Reply:
Thank you. I have initialized it with nullptr and added AMREX_RESTRICT

Reply:
You can also call `const_array` instead of `array` for data that are read only.  You can still use `auto const&` for the return type.  The acutal type is `Array4<Real const>`.   This is kind of like using `intent(in)` in Fortran.

Reply:
Thank you. I have made these changes now to the definition of the sigma arrays.